docs/notes/migration_guide.rst

```diff
@@ -0,0 +1,491 @@
+      * - ``fastrtps/transport/UDPv6TransportDescriptor.h``
+        - ``fastdds/rtps/transport/UDPv6TransportDescriptor.hpp``
+      * - ``fastrtps/transport/UDPTransportDescritpor.h``
+        - ``fastdds/rtps/transport/UDPTransportDescritpor.hpp``
+      * - ``fastrtps/transport/TCPTransportDescritpor.h``
+        - ``fastdds/rtps/transport/TCPTransportDescritpor.hpp``
+      * - ``fastdds/rtps/common/Time_t.hpp in namespace{fastdds}``
+        - ``fastdds/dds/core/Time_t.hpp in namespace{fastdds::dds}``
+
+   Also, the ``fixed_size_string.hpp`` implementation has been migrated from Fast DDS package to Fast CDR.
+
+2. File Extensions:
+
+   Rename file extensions from `.h` to `.hpp`.
+
+.. _step-6-handle-removed-or-private-headers:
+
+Step 6: Handle Removed or Private Headers
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following list contains headers that were previously in the `include` folder and have been relocated to the `src/cpp` folder.
+Since they are no longer public, it is not possible to include them in external projects:
+
+* ParticipantAttributes.hpp
+* ReplierAttributes.hpp
+* RequesterAttributes.hpp
+* PublisherAttributes.hpp
+* SubscriberAttributes.hpp
+* ProxyPool.hpp
+* Semaphore.hpp
+* MessageReceiver.hpp
+* BuiltinProtocols.hpp
+* shared_mutex.hpp
+* StringMatching.hpp
+* TimeConversion.hpp
+* DBQueue.hpp
+* ResourceEvent.hpp
+* TimedEvent.hpp
+* WriterProxyData.hpp
+* ReaderProxyData.hpp
+* ParticipantProxyData.hpp
+* XML Parser API
+* UnitsParser.hpp
+* RTPSMessageGroup.hpp
+* RTPSMessageCreator.hpp
+* CDRMessage.hpp
+* StatefulPersistentReader.hpp
+* StatefulReader.hpp
+* StatelessPersistentReader.hpp
+* StatelessReader.hpp
+* PersistentWriter.hpp
+* StatefulPersistentWriter.hpp
+* StatefulWriter.hpp
+* StatelessPersistentWriter.hpp
+* StatelessWriter.hpp
+* logging.h
+* Exception.h
+* Cryptography.h
+* Authentication.h
+* AccessControl.h
+* SecurityException.h
+* ChangeForReader.hpp
+* ReaderLocator.hpp
+* ReaderProxy.hpp
+* ServerAttributes.hpp
+* TopicAttributes.hpp
+* TypeLookupService.hpp
+
+If your project previously included any of these headers, you will need to modify your implementation.
+Since these headers are now private, you should replace their usage with public alternatives or refactor the related code to ensure it does not depend on private headers.
+
+.. _step-7-update-api-methods:
+
+Step 7: Update API Methods
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The table below contains the list of API changes, showing the previous methods and the corresponding new ones introduced in Fast DDS v3.
+The new API methods achieve the same functionality, even though the signature of the method is different from the deprecated one.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Deprecated methods
+     - New methods
+   * - ``xmlparser::XMLProfileManager::library_settings(LibrarySettingsAttributes&)``
+     - ``DomainParticipantFactory::get_instance()->set_library_settings(const LibrarySettings&)``
```

Comment:
Ideally, we should reference the API reference in the new methods, but it is a lot of work...

Reply:
But I wouldn't be able to reference the deleted methods, right?

Reply:
Exactly

Reply:
This will be done in a follow-up PR. See [21797] (https://eprosima.easyredmine.com/issues/21797)